staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -75,6 +78,14 @@ var (
 		},
 		[]string{"resource"},
 	)
+	resourceSizeEstimate = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "apiserver_resource_size_estimate_bytes",
+			Help:           "Estimated size of stored objects in database. Estimate is based on sum of last observed sizes of serialized objects. In case of a fetching error, the value will be -1.",
```

Comment:
@richabanker is the use of -1 here reasonable from a metrics conventions perspective?  I'm not familiar enough to with the metrics conventions to give guidance here.

Reply:
thats a good call out, @serathius why do we need to expose a -1 in case of errors? That could mess with aggregation for this metric no? Cant we just not record the metric in case of errors?

Reply:
Filled https://github.com/kubernetes/kubernetes/issues/132989

Reply:
Context https://github.com/kubernetes/kubernetes/pull/59757#discussion_r169251531 
/cc @liggitt

Reply:
Hmm Prometheus2.0 has added some improvements around staleness of a metric looks like, [ref](https://promcon.io/2017-munich/slides/staleness-in-prometheus-2-0.pdf) so maybe absent timeseries are treated correctly as stale.

Anyway since the -1 value already existed before, and we have had no complaints about its usability so far, I guess its fine to have this as it is and discuss the staleness / not-recording-a-metric-when-error topic in the dedicated issue, thanks for creating it!

Reply:
/lgtm